### PR TITLE
Merged in masters mbb changes and removed depreciations

### PR DESCRIPTION
**Pull request by lwahlmeier**

### CODE REVIEW

File: src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java [hunk 1]
```diff
@@ -145,16 +144,13 @@ public void stopListening(final Server server) {
     if(!checkServer(server)) {
       return;
     } else {
-      if(server.getServerType() == WireProtocol.TCP) {
+      if(server instanceof TCPServer) {
```
Comment (jentfoo) [inline src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java]: The above check should have been more efficient than this now instanceof check.  Why the change?

File: src/main/java/org/threadly/litesockets/utils/IOUtils.java [hunk 2]
```diff
@@ -182,18 +193,21 @@ public int read(byte[] ba, int offset, int len) throws IOException {
     
     @Override
     public int read() throws IOException {
+      if(isClosed) {
+        return -1;
+      }
       while(true) {
-        synchronized(this) {
+        synchronized(currentBB) {
```
Comment (jentfoo) [inline src/main/java/org/threadly/litesockets/utils/IOUtils.java]: It seems like you should synchronized before the while loop to prevent releasing and then immediately re-acquiring the lock as you loop

File: src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java (hunk 1 above)
Comment (lwahlmeier) [inline src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java]: its more correct esp with the added casting below, and honestly the performance difference is pretty minimal.  This is also only ever called if you call .stop() on a server instance so I dont see a practical use case where performance is gonna matter a lot there.

File: src/main/java/org/threadly/litesockets/utils/IOUtils.java (hunk 2 above)
Comment (lwahlmeier) [inline src/main/java/org/threadly/litesockets/utils/IOUtils.java]: good point
